framer.go

```diff
@@ -82,6 +94,29 @@ func (f *framer) AppendControlFrames(frames []ackhandler.Frame, maxLen protocol.
 		}
 	}
 
+	// add stream-related control frames
+	for id, str := range f.streamsWithControlFrames {
+	start:
+		remainingLen := maxLen - length
+		if remainingLen <= minStreamControlFrameSize {
+			break
+		}
+		fr, ok, hasMore := str.getControlFrame()
+		if !hasMore {
+			delete(f.streamsWithControlFrames, id)
+		}
+		if !ok {
+			continue
+		}
+		frames = append(frames, fr)
+		length += fr.Frame.Length(v)
+		if hasMore {
+			// It is rare that a stream has more than one control frame to queue.
+			// We don't want to spawn another loop for just to cover that case.
+			goto start
+		}
+	}
+
 	for len(f.controlFrames) > 0 {
 		frame := f.controlFrames[len(f.controlFrames)-1]
```

Comment:
As we pop from the back of the slice, would this mean that we send `MAX_STREAMS_FRAME` in reverse order higher before lower? This is not a protocol violation so it's fine. I'm just curious.

Reply:
Possibly! Another reason to rewrite this code path: We should only ever send a single MAX_STREAMS frame (the one with the highest value).